pyro/generic/distributions.py

```diff
@@ -0,0 +1,5 @@
+from pyro.distributions import *  # noqa: F401, F403
+
+# Expose constraints and transforms
+from torch.distributions.constraints import *  # noqa: F401, F403
```

Comment:
I think it's a bit safer to access constraints and transforms in a partially qualified way, e.g. `transforms.Planar` and `constraints.positive`. Could we allow this somehow, say either `import torch.distributions.constraints as constraints` or treat these as a top-level `pyro_generic`-managed thing?

Note this is becoming more important as #1932 is dropping `Transform` suffix on transforms and assumes they will be accessed partially qualified.

Reply:
Hmm..either option sounds good to me. I'll go with the first one right now, but let me know if you would prefer it to be specified at the generic level.

Reply:
The first option unfortunately doesn't work because we have a local `pyro.distributions.transforms` module, so I cannot just import `torch.distributions.transforms` into `pyro.distributions.__init__`. I thought it will be best to go with the second option instead which is creating separate generic modules for constraints and transforms.

Reply:
I think it might make most sense to import torch transforms into pyro transforms, so that just like pyro.distributions is a strict extension of torch.distributions, pyro.distributions.transforms would be a strict extension of torch.distributions.transforms:
```py
# in pyro/distributions/transforms.py
from torch.distributions.transforms import *
```
@stefanwebb WDYT?

Reply:
I like the idea of adding `constraints`  and `transforms` to `pyro.generic`. I think we'll also need a bit of reorg #2052 so that both pyro and torch transforms are accessible.

Reply:
Done. I have also imported `torch.constraints` into the namespace which can be accessed as `pyro.distributions.constraints`.

Reply:
Sorry, I missed your last comment. The latest commit assumes that `transforms` and `constraints` are available from the distributions namespace. I am not sure if that conflicts with any plans from #2052. I am happy to revert that to the previous choice of exposing these directly from `pyro.generic`.

Reply:
Just looking at #2052 - lets get that merged first.

Reply:
Now that #2052 already exposes `transforms` and `constraints`, do you have any opinion on whether we should also separately expose them in `generic`

```
'transforms': 'pyro.distributions.transforms'
```

 or just assume that they will be available from the `distributions` namespace for generic dispatch?

Reply:
I think it's a little cleaner to expose `constraints` and `transforms` from generic.

Reply:
On second thought, I have no opinion. I believe this PR is mostly orthogonal from #2052 : in either one we need to be able to access both torch and pyro transforms in a single place.

Reply:
Sounds good. I'll just merge this against master but keep things as they are, for now.